ANTLR/src/Compiler/SymbolTable/SymbolDefListener.java

```diff
@@ -0,0 +1,54 @@
+package Compiler.SymbolTable;
+
+import Compiler.Lang.LangBaseListener;
+import Compiler.Lang.LangParser.*;
+import org.antlr.v4.runtime.ParserRuleContext;
+import org.antlr.v4.runtime.tree.ParseTreeProperty;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class SymbolDefListener extends LangBaseListener{
+    public ParseTreeProperty<Scope> scopes = new ParseTreeProperty<Scope>();
+    public BaseScope globalScope = new BaseScope();
+    Scope currentScope;
+
+    @Override
+    public void enterProg(ProgContext ctx) {
+        currentScope = globalScope;
+    }
+
+    @Override
+    public void enterFuncdef(FuncdefContext ctx) {
+        //Gets lists of parameters as ParseRuleContexts
+        List<FuncdefparamContext> params =  ctx.getRuleContext(FuncdefparamsContext.class, 0)
+                .getRuleContexts(FuncdefparamContext.class);
+
+        //Might be usefull for type-checking. Delete otherwise
+        List<Type> argumentList = new ArrayList<Type>();
```

Comment:
It is a bit confusing that our Type is called Type. As one might think that Type is a generic keyword. But it is actually a class in our implementation.